### ISSUE TITLE
Is the .vscale needed in chebtech objects?

### ISSUE THREAD

**Issue by nickhale** [chebfun/chebfun#1479]
Since it seems the `.epslevel` property of `chebtech` objects is about to go the way of the Dodo, should we also consider scrapping the `.vscale` field? 

In terms of efficiency, I imagine around 50% of the time we will lose by having to compute the same vscale more than once, and the other 50% we will save by having not computed it at all.

In terms of the code, it would almost certainly make things simpler: `vscale()` would become a method rather than a property and only called when the vertical scale is required. Strictly speaking, the vscale can always be recovered from the `.coeffs`, so storing it is a duplication of information.

I think the only reason we _have_ a `.vscale` currently is a hang-over from v4, where it represented something other than the current magnitude of the chebfun.

**Comment by trefethen** [comment]
A very interesting idea.  @jaurentz and I will bear it in mind as we work further on matters of constructors etc.  Thanks.

**Comment by jaurentz** [comment]
Right now we've only disabled `epslevels`, with plans for removal in the future. Is a straight forward thing to disable `vscales` before completely removing it?

**Comment by aaustin141** [comment]
@jaurentz:  Just to be clear, the proposal is not to eliminate the `vscale` concept (like we've more or less done with `epslevel`; we've just kept the code lying around because we may try to reintroduce it in the not-so-distant future).  It's just to compute `vscale` only on-demand instead of updating it at the end of every operation and storing the result.

I'm generally in favor of doing this, assuming it doesn't too negatively impact performance.  Getting rid of all the code required to update it would definitely simplify things and eliminate improperly updated `vscale` fields from our list of sources of bugs.  On the other hand, `vscale` is probably the most frequently requested property of a tech, so we should measure some experimental timings first before we commit to doing this.

**Comment by nickhale** [comment]
> On the other hand, vscale is probably the most frequently requested property of a tech, so we should measure some experimental timings first before we commit to doing this.

But since accessing properties is so slow in MATLAB, we might actually _gain_ from this..

**Comment by trefethen** [comment]
I very much support the idea: I like simplicity.

**Comment by aaustin141** [comment]
>> On the other hand, vscale is probably the most frequently requested property of a tech, so we should measure some experimental timings first before we commit to doing this.

> But since accessing properties is so slow in MATLAB, we might actually _gain_ from this..

I'm not contradicting that; to the contrary, I think this is probably true.  I'm just saying that we should measure to be sure.

**Comment by nickhale** [comment]
>>> On the other hand, vscale is probably the most frequently requested property of a tech, so we should measure some experimental timings first before we commit to doing this.

>> But since accessing properties is so slow in MATLAB, we might actually _gain_ from this..

>I'm not contradicting that; to the contrary, I think this is probably true.  I'm just saying that we should measure to be sure.

OK - cool.

**Comment by nickhale** [comment]
I decided to take a quick look at this. It seems pretty simple.

On question that arises is, what to call the method which now returns the vscale? 

Previously `.vscale` was a property, so we weren't allowed a `vscale()` method. Instead we have a `getVscale()` method. 

Now `.vscale` is gone, `vscale()` seems the obvious name for the method. However, a lot of the time we used `vscale` as a variable in the chebtech methods, so this now causes a conflict. 

In summary, two options:
 * Continue to use the `getVscale()` method and use `vscale` as a variable.
 * Use `vscale()` as the method and forbid the use of `vscale` within chebtech methods (i.e., use `vscl` instead).

Any opinions? I have a slight preference for the latter since `vscale()` exists as a (hidden) `chebfun` method.

**Comment by nickhale** [comment]
I also notice that a **lot** of the `vscale` stuff is just being used to determine an updated `.epslevel`. If `epslevels` get totally removed then things become even more transparent.

**Comment by jaurentz** [comment]
From what I've seen so far, it appears that the change to `vscales` and removal of `epslevels` needs to be done simultaneously. I am all for this but I need to know how the `vscale` change should be handled before I can proceed.

**Comment by aaustin141** [comment]
Closed in #1500.
